src/main/java/com/cloudant/client/api/CloudantClient.java

```diff
@@ -484,10 +503,10 @@ Response put(URI uri, Object object, boolean newEntity, int writeQuorum, Gson gs
                 assertNotEmpty(id, "id");
                 assertNotEmpty(rev, "rev");
             }
-            final HttpPut put = new HttpPut(buildUri(uri).pathToEncode(id).query("w",
-                    writeQuorum).buildEncoded());
-            CouchDbUtil.setEntity(put, json.toString(), "application/json");
-            response = client.executeRequest(put);
+            HttpConnection putConnection = Http.PUT(buildUri(uri).pathToEncode(id).query("w",

```

Comment:
Partial fix in e6dd315.  Left a note for tomorrow to discuss if we should move the method into `CouchDbClient` class.